### PR TITLE
Follow up on #18424 comments

### PR DESCRIPTION
**Pull request by xster**
Follow up on #18424 comments

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

File: packages/flutter/lib/src/painting/image_stream.dart [hunk 1]
```diff
@@ -242,26 +242,28 @@ abstract class ImageStreamCompleter extends Diagnosticable {
...
   /// its associated [ImageErrorListener].
   void removeListener(ImageListener listener) {
-    _listeners.removeWhere((_ImageListenerPair listenerPair) {
-      return listenerPair.listener == listener;
-    });
+    _listeners.remove(
+      _listeners.firstWhere((_ImageListenerPair listenerPair) {
+        return listenerPair.listener == listener;
+      })
+    );
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/image_stream.dart]: i suspect this is expensive relative to just walking the list and removing the first match by index.

**Review by Hixie — COMMENTED** [review]
LGTM

File: packages/flutter/lib/src/painting/image_stream.dart [hunk 2]
```diff
@@ -166,9 +164,12 @@ class ImageStream extends Diagnosticable {
     if (_completer != null)
       return _completer.removeListener(listener);
     assert(_listeners != null);
-    _listeners.removeWhere((_ImageListenerPair listenerPair) {
-      return listenerPair.listener == listener;
-    });
+    for (int i = 0; i < _listeners.length; ++i) {
+      if (_listeners[i].listener == listener) {
+        _listeners.removeAt(i);
+        continue;
```
Comment (kgdev) [inline packages/flutter/lib/src/painting/image_stream.dart]: this should be break?

Comment (xster) [reply packages/flutter/lib/src/painting/image_stream.dart]: Thanks https://github.com/flutter/flutter/pull/25994

**Pull request merged**